src/v/redpanda/debug_bundle.cc

```diff
@@ -269,3 +272,21 @@ ss::future<std::unique_ptr<ss::http::reply>> debug_bundle::fetch_bundle(
     co_return co_await _transfer_handle.handle(
       {}, std::move(req), std::move(rep));
 }
+
+ss::future<> debug_bundle::remove_bundle(ss::sstring& filename) {
+    if (ss::this_shard_id() != debug_bundle_shard_id) {
+        co_await container().invoke_on(
+          debug_bundle_shard_id,
+          [&filename](debug_bundle& b) { return b.remove_bundle(filename); });
+    }
+
+    debug_bundle_status bundle_status{debug_bundle_status::not_running};
+    if (filename == _in_progress_filename) {
+        bundle_status = co_await get_status();
+    };
+    detail::throw_if_bundle_dne(bundle_status, get_write_dir(), filename);
+
+    co_await ss::remove_file(
+      detail::make_bundle_filename(get_write_dir(), filename));
+    co_await ss::sync_directory(get_write_dir().c_str());
```

Comment:
So here, we are permitting the user to delete whatever file exists in `write_dir`.  We should ensure we only allow deletions of actual bundle files.

Reply:
Maybe I misunderstand something, but that is the reason for calling `detail::throw_if_bundle_dne`. That method will throw if the user supplies a filename that is not actually a bundle.

Reply:
So I ran a test locally and can confirm that the `DELETE` handler will delete any file in the `write_dir`

Reply:
Fair point. That check should also be for all requests that have `{filename}` in the path as well.